core/trino-spi/src/main/java/io/trino/spi/predicate/Range.java

```diff
@@ -45,11 +45,14 @@
     private final boolean isSingleValue;
 
     Range(Type type, boolean lowInclusive, Optional<Object> lowValue, boolean highInclusive, Optional<Object> highValue)
+    {
+        this(type, lowInclusive, lowValue, highInclusive, highValue, getComparisonOperator(type));
+    }
+
+    Range(Type type, boolean lowInclusive, Optional<Object> lowValue, boolean highInclusive, Optional<Object> highValue, MethodHandle comparisonOperator)
```

Comment:
Now that this is a constructor parameter, add `requireNonNull` check

Reply:
done